### PR TITLE
Added neotree-open-file-in-system-application

### PR DESCRIPTION
**Pull request by Grimscythe1983**
I thought that this would be a nice addition to this excellent package. Dired has dired-launch, NeoTree needs one too!

### CODE REVIEW

File: neotree.el [hunk 1]
```diff
@@ -1829,6 +1836,14 @@ FULL-PATH and ARG are the same as `neo-open-file'."
 FULL-PATH and ARG are the same as `neo-open-file'."
   (neo-open-file full-path "a"))
 
+(defun neotree-open-file-in-system-application ()
+  "Open a file under point in the system application."
+  (interactive)
+  (shell-command
+   (concat neo-default-system-application
+           " "
+           (neo-buffer--get-filename-current-line))))
```
Comment (omajid) [inline neotree.el]: Should this be quoted with `shell-quote-argument` or `combine-and-quote-strings`?

Comment (Grimscythe1983) [inline neotree.el]: Maybe? However...it works as it is. So I am not sure what would be the difference.

Comment (Grimscythe1983) [inline neotree.el]: I read up on the functions that you have mentioned and it does make sense to modify the code. Ill update it as soon as possible.

**Comment by Grimscythe1983** [comment]
I decided to change the call to `call-process`, the solution proposed previously by @omajid worked, yes! However I thought that creating a separate process via `call-process` makes much more sense and it also has less quirks than calling `async-shell-command`

**Pull request merged**